### PR TITLE
Glossary Terms

### PR DESCRIPTION
**Pull request by darrellodonnell**
I am working to get some closure on definitions and terms that should help simplify a few things. The work is not complete, but this PR starts aims to show what needs to be done (and completes a few things).

One item that popped up as a surprise was that `spec-up` was added to `package.json` and `package-lock.json`. I think that is ok, but I was surprised that it wasn't already there. I suppose nobody is doing builds locally.

### CODE REVIEW

File: spec/appendix.md
Comment (swcurran) [inline spec/appendix.md:6]: It’s not clear what definitions “need to be moved” — the ones in this spec or the ones in the ToIP Glossary.  I think you mean that the “ToIP Glossary definitions are being moved to the (under development) ToIP Terms tool (aka TEv2). 

Suggest you clarify.

File: spec/appendix.md
Comment (swcurran) [inline spec/appendix.md:7]: Presumably you mean “not all the relevant ToIP definitions…”.  Presumably, the ToIP Glossary is much bigger than this.

File: spec/appendix.md
Comment (darrellodonnell) [reply spec/appendix.md:6]: That's going to be an ongoing effort as we edit this (and other specs).

File: spec/appendix.md
Comment (swcurran) [inline spec/appendix.md:13]: Scary to go to this level in a spec of this type. Do you really think this is needed?  That said, if it is needed and you are putting a Source, I suggest including the link to the source.

Comment (darrellodonnell) [reply spec/appendix.md:13]: It's a good example of a definition/term that can be left out of this spec and point at the ToIP Glossary. Drummond and I are working with Henk to get the tooling figured. For now I'd prefer to leave it in as we don't have a place to point it. 

This is the entry in the (Google Docs) doc that is a temporary place. 

image

File: package.json
Comment (swcurran) [inline package.json:58]: Hmm…seems weird this is needed. I’ve not done it in a while, but have definitely tested locally and it had been working fine.  Doesn’t seem like it was ever removed.  And the GHA publishing is working fine...

Comment (darrellodonnell) [reply package.json:58]: Kevin recently added the spec-up tooling.

File: spec/appendix.md
Comment (darrellodonnell) [reply spec/appendix.md:6]: relates to this Issue: https://github.com/trustoverip/tswg-did-method-webs-specification/issues/116

File: spec/appendix.md
Comment (swcurran) [reply spec/appendix.md:13]: I would suggest for all of these “defined in the ToIP Glossary” terms, you put the link to the ToIP Glossary.  E.g. for this one use [this link](https://docs.google.com/document/d/1fZByfuSOwszDRkE7ARQLeElSYmVznoOyJK4sxRvJpyM/edit#heading=h.53rzpn1yn6q7) and leave out the reference to Source being Wikipedia.

Or if you don’t want to link to exact header, at least the link to the ToIP Glossary, so we know all the things to be fixed later...

Comment (darrellodonnell) [reply spec/appendix.md]: resolved in recent push

File: package.json
Comment (darrellodonnell) [reply package.json]: removed this in https://github.com/trustoverip/tswg-did-method-webs-specification/pull/117/commits/6cd709751e077208ecee9cdbff527438f3c9ff74

File: spec/appendix.md
Comment (2byrds) [inline spec/appendix.md:34]: Is the idea to wait for this work to be completed before we merge this? For instance we generally would only merge a PR that adds terminology if the terminology has [[ref:]] updates that use the [[def:]] terminology being added. And I think the TODO section shouldn't be merged but could be converted into a set of issues that we could process over time. We could decide to add some to the Spec 1.0 milestone or not.

**Review by 2byrds — CHANGES_REQUESTED** [review]

**Comment by darrellodonnell** [comment]
I would prefer to get the work in now, not wait.

Darrell O'Donnell

***@***.***

[image: __tpx__]

File: spec/appendix.md
Comment (darrellodonnell) [reply spec/appendix.md:34]: I would prefer to get the work in now, not wait.

There are enough "things" that need to be cleaned up that we should start immediately

Comment (2byrds) [reply spec/appendix.md:34]: From our meeting today:
@darrellodonnell will break these Todos out as issues and we'll decide which belong in 1.0 and 1.1 spec milestones.

Comment (darrellodonnell) [reply spec/appendix.md:34]: removed the todo items.

File: package-lock.json
Comment (2byrds) [inline package-lock.json:41]: @darrellodonnell this is what i referred to in the slack message

File: package-lock.json
Comment (2byrds) [inline package-lock.json:3571]: This as well.

**Review by 2byrds — CHANGES_REQUESTED** [review]
specup related changes should be removed.

File: package-lock.json
Comment (darrellodonnell) [reply package-lock.json:3571]: ok - removed these (unnecessary) changes to package-lock.json.

File: package-lock.json
Comment (darrellodonnell) [reply package-lock.json:41]: removed

File: spec/appendix.md
Comment (2byrds) [reply spec/appendix.md:6]: @darrellodonnell I believe this should be moved out into a separate issue and PR (its basically a TODO)

File: spec/appendix.md
Comment (2byrds) [reply spec/appendix.md:7]: @darrellodonnell I believe this TODO line should be removed. The addition of any other terms can be included in a future PR.

Comment (darrellodonnell) [reply spec/appendix.md:7]: noted - I have made edits to remove the NOTE as well as removed a few definitions that go too far (i.e. really belong in an external glossary). Added issue to GH as well: https://github.com/trustoverip/tswg-did-method-webs-specification/issues/121 

update coming shortly

**Review by 2byrds — APPROVED** [review]
LGTM! Thank you @darrellodonnell !

File: spec/appendix.md
Comment (swcurran) [inline spec/appendix.md:15]: I think the [[def: crypto…] should be a a [[ref: …], correct?

**Review by swcurran — CHANGES_REQUESTED** [review]
A quick tweak and this looks ready.

File: spec/appendix.md
Comment (darrellodonnell) [reply spec/appendix.md]: good catch. Caused me to remove a few of the refs that I had removed defs for (e.g. third party, verification). change committed.

**Review by swcurran — APPROVED** [review]

**Pull request merged**
